src/librustdoc/core.rs

```diff
@@ -213,7 +217,7 @@ pub fn run_core(search_paths: SearchPaths,
         } else {
             vec![]
         },
-        lint_cap: Some(lint::Forbid),
+        lint_cap: Some(lint_cap.unwrap_or_else(|| lint::Forbid)),
```

Comment:
You can leave this as `None` if you're going to default it to `Forbid` anyway; rustc will do this unwrap itself later on.

Reply:
No, I did at first but didn't get the expected result.

Reply:
Huh, weird, I didn't realize it was unwrapped anywhere other than where it was used.